### PR TITLE
Search and sort_link added using ransack

### PR DESCRIPTION
**Pull request by tarunjadhwani**
**What this PR does:**
- Added search form on the navbar
- Added sort_link for various article attributes

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -2,7 +2,8 @@ class ArticlesController < ApplicationController
   before_action :authenticate_user!, only: [:index, :show, :new, :edit, :create, :update, :destroy]
 
   def index
-    @articles = Article.all
+    @q = Article.ransack(params[:q])
```
Comment (tarunjadhwani) [inline app/controllers/articles_controller.rb:5]: not needed.

File: app/views/articles/_list.html.haml [hunk 2]
```diff
@@ -1,13 +1,17 @@
 %table.my-table
   %tr
-    %th Title
+    %th= sort_link @q, :title
     %th Text
+    %th= sort_link @q, :created_at, default_order: :desc
```
Comment (tarunjadhwani) [inline app/views/articles/_list.html.haml:6]: use one. Try this is even working.

Comment (coolRaider) [inline app/views/articles/_list.html.haml:6]: this is working, first time when we click on it, it sorts in descending order, for title, it sorts in ascending order when clicked the first time as that's the default behaviour.

File: app/controllers/articles_controller.rb (hunk 1 above)
Comment (coolRaider) [inline app/controllers/articles_controller.rb:5]: not applicable anymore as moved search form to index page from navbar.

**Pull request merged**